firmware/ventilator-controller-stm32/Core/Inc/Pufferfish/Util/COBS.tpp

```diff
@@ -115,7 +116,7 @@ constexpr size_t get_encoded_cobs_buffer_size(
       write_index++;
     }
     if (read_index > unencoded_buffer.size()) {
```

Comment:
What if `read_index == unencoded_buffer.size()`?

Reply:
I think read_index is always less than `unencoded_buffer.size()` as it won't enter the while loop if it's equal to it. So in a way we are implicitly doing the bounds check, then the question is do we need to explicitly catch `out_of_bounds` ?

Reply:
Oops!!! I had been thinking of this function as reading encoded data, when clearly it's reading pre-encoded data. The only way we read data is checking if a byte is zero, and deciding based on that whether to increment some (index) counters. So the bounds check of the while loop is sufficient; invalid data has no way to force an out-of-bounds read.